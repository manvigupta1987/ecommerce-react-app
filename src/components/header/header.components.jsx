import React from 'react';
import './header.styles.scss'
import {Link} from 'react-router-dom'
import {ReactComponent as Logo} from '../../assets/logo.svg'
import {auth} from '../../firebase/firebase.utils';
//Connect is a HOC 
import {connect} from 'react-redux';

const Header = ({currentUser}) => {
  return(
    <div className='header'>
      <Link className='logo-container' to='/'>
        <Logo className='logo'/>
      </Link>
      <div className='options'>
        <Link className='option' to='/shop'>SHOP</Link>
        <Link className='option' to='/contact'>CONTACT</Link>
        { currentUser ?
          <div className='option' onClick={()=> auth.signOut()}>SIGN OUT</div> 
          : 
          <Link className='option' to='/signin'>SIGN IN</Link>
        }
        
      </div>
    </div>
  );
}
// name of this function can be anything but mapStateToProps is standard with redux codebases. 
// the value that is recived in this is the state object which is root reducer. 
// this function will be used anywhere we need property from the reducer. 
const mapStateToProps = (state) => ({
  currentUser: state.user.currentUser
})

//connect is the high order component that takes a component as an argument and returns a component. 
export default connect(mapStateToProps)(Header);