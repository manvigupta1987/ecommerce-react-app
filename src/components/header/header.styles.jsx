// import styled, {css} from 'styled-components';
import styled from 'styled-components';
import {Link} from 'react-router-dom';

// const OptionContainerStyles = css`
//   padding: 10px 20px;
//   cursor: pointer;
// `
export const HeaderContainer = styled.div`
  width: 100%;
  height: 70px;
  display: flex;
  justify-content: space-between;
  margin-bottom: 25px;
  padding: 0 80px;
`;
export const LogoContainer = styled(Link)`
  width:70px;
  height: 100%;
  padding: 25px;
`;

export const OptionsContainer = styled.div`
  width:50%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
`

// export const OptionLink = styled(Link)`
//   ${OptionContainerStyles}
// `

// export const OptionDiv = styled.div`
//   ${OptionContainerStyles}
// `


export const OptionLink = styled(Link)`
  padding: 10px 20px;
  cursor: pointer;
`