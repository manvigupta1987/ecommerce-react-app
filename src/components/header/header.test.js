import React from 'react';
import { shallow } from 'enzyme';
import {Header} from './header.components';
import CartDropDown from '../cart-dropdown/cart-dropdown.component';

describe('Header Component', ()=> {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      currentUser : {
        uid: '123'
      },
      hiddenCart: true
    }
    wrapper = shallow(<Header {...mockProps} />)
  })

  it('header component snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('expect to render sign out text when user is present', ()=> {
    expect(wrapper.find('OptionLink').at(2).text()).toEqual('SIGN OUT');
  })

  it('expect to render sign in text when user is null', ()=> {
    const mockProps1 = {
      currentUser : null,
      hiddenCart: true
    }
    const wrapper2 = shallow(<Header {...mockProps1} />)
    expect(wrapper2.find('OptionLink').at(2).text()).toEqual('SIGN IN');
  })

  it('expect to render nothing when hidden is true', ()=> {
    expect(wrapper.exists(CartDropDown)).toBe(false);
  })

  it('expect to render CartDropDown when hidden is false', ()=> {
    const mockProps1 = {
      currentUser : null,
      hiddenCart: false
    }
    const wrapper2 = shallow(<Header {...mockProps1} />)
    expect(wrapper2.exists(CartDropDown)).toBe(true);
  })



})