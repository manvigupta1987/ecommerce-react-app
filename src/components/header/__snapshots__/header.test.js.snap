// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component header component snapshot testing 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    <ForwardRef(SvgLogo)
      className="logo"
    />
  </Styled(Link)>
  <styled.div>
    <OptionLink
      to="/shop"
    >
      SHOP
    </OptionLink>
    <OptionLink
      to="/contact"
    >
      CONTACT
    </OptionLink>
    <OptionLink
      as="div"
      onClick={[Function]}
    >
      SIGN OUT
    </OptionLink>
    <Connect(CartIcon) />
  </styled.div>
</styled.div>
`;
